### ISSUE TITLE
Add "Intro to Wavefunction" Documentation

### ISSUE THREAD

**Issue by JonathonMisiewicz** [psi4/psi4#2339]
Recent forum interactions have exposed that even if the wavefunction class were documented, there's no page that tells users "to find quantities, look on the wavefunction." The importance of the wavefunction doesn't seem documented anywhere. This should be added to the Psi4 tutorial section, possibly as a new page on how to find quantities in Psi4.

**Comment by loriab** [comment]
Agreed. https://github.com/psi4/psi4numpy/blob/master/Tutorials/01_Psi4NumPy-Basics/1d_wavefunction.ipynb is where we usually send folks for wfn info, but that's not self-discoverable.

**Comment by VHchavez** [comment]
I could work on this :)
It would help me to have some broad ideas that the docs need to cover and whether they should be described in terms of Psi4 or Psi4API.

**Comment by hokru** [comment]
I think one basic example should show psithon and Psi4API together and point to the difference.
Then the rest maybe in Psi4API style, because the more advanced things are usually done in python scripts, at least looking at the Forum.

How to retrieve and write data to a wavefunction is a common issue.

**Comment by loriab** [comment]
Thanks, @VHchavez, that'd be a great help. I think PsiAPI is the better framework to describe more advanced topics like wfn. (The exception is the two mirrored tutorials in PSIthon and PsiAPI that we try to keep in sync.) Topics include how to get wfn, things on wfn, things not on wfn (e.g., amplitudes in most cases), when wfn might be inconsistent (energy is mp2, while orbitals are scf), numpy serialization (python_helpers.py), schema serialization (schema_wrapper.py), etc. Another thing is feel free to be more of a clearinghouse where you collect links to the p4n tutorial, the serialization code, the psi4.core.Wavefunction API docs, various test cases that use things off the wfn. All those other places are autogen or tested, so are a better place than docs for big code snippets. And folks interested in wfn are usually able and willing to read python code if directed, just not willing to sift through 1M lines to find it.

**Comment by JonathonMisiewicz** [comment]
I agree with Holger about Psithon vs PsiAPI: work primarily in PsiAPI, but translate an input file to Psithon to show you _can_.

Lori's topic list is a good one. Explicitly mention how to get orbitals and how to get variables (like the correlation energy) off the wavefunction.
